src/distributions/weighted_index.rs

```diff
@@ -141,6 +141,10 @@ impl<X: SampleUniform + PartialOrd> WeightedIndex<X> {
     /// allocation internally.
     ///
     /// In case of error, `self` is not modified.
+    /// 
+    /// Note: Updating floating-point weights may cause slight inaccuracies in the total weight.
+    ///           This method will not always return `WeightedError::AllWeightsZero` when all weights
+    ///           are zero if using floating-point weights. 
```

Comment:
```suggestion
```suggestion
    /// Note: Updating floating-point weights may cause slight inaccuracies in the total weight.
    ///       This method may not return `WeightedError::AllWeightsZero` when all weights
    ///       are zero if using floating-point weights. 
```